packages/cdktf/lib/terraform-resource.ts

```diff
@@ -231,4 +252,64 @@ export class TerraformResource
       this.cdktfStack
     );
   }
+
+  private parentStackResourceTargets() {
+    return TerraformStack.of(this).moveTargets;
+  }
+
+  /**
+   * Moves this resource to the location denoted by moveTarget.
+   * @param moveTarget The previously set string moveTarget denoting the location of the move
+   * @param index Optional The index corresponding to the key the resource is to appear in the foreach of a resource to move to
+   */
+  public moveTo(moveTarget: string, index?: string | number) {
+    const stackMoveTargets = this.parentStackResourceTargets();
+    const resourceToMoveTo =
+      stackMoveTargets.getResourceAddressByTarget(moveTarget);
```

Comment:
Have you tested if there are cases where `moveTo` is evaluated before `addMoveTarget`? E.g. by changing the order of calls to the resource and to the nested construct in the main stack? I suspect we might need to decouple the evaluation here (I can prototype something, might need an aspect to delay validation until all constructs are added).